helper/chksum.c

```diff
@@ -280,11 +280,12 @@ int odph_udp_tcp_chksum(odp_packet_t     odp_pkt,
 
 	/* First parse and process the l4 header */
 	rc = odph_process_l4_hdr(odp_pkt, op, &udp_tcp_hdr, chksum_ptr, &l4_len,
-				 &split_l4_hdr, &is_tcp, &pkt_chksum_offset,
-				 &pkt_chksum_ptr);
+				 &split_l4_hdr, &is_tcp, &pkt_chksum_offset);
 	if (rc != 0)
 		return rc;
 
+	pkt_chksum_ptr = (uint16_t *)(((uint8_t *)odp_packet_data(odp_pkt)) + pkt_chksum_offset);
```

Comment:
This loses information about the minimum guaranteed aligment of the checksum field (which is 1). pkt_chksum_ptr is uint16_t pointer which has guaranteed alignment of 2. The cast through uint8_t* hides the problem in most platforms (except if --Wcast-align=strict is used, I think).

The overall approach for not taking address of a packed struct member looks ok to me, except for this line which does basically the same potentially dangerous thing but in a different way to hide it from view.

My point is that if we wanted to silence the warning without really fixing anything, we could have just added a suitable pragma and be done with it. But if we want to silence the warning by getting rid of the alignment issue that causes it, then it is not good to introduce another, similar but subtly different, alignment issue in the code as part of the fix (even if there may be other alignment problems, not warned about by the compiler, in the existing code).

Reply:
Fixed